app/src/context/evm/prompt-wars-market-contract/PromptWarsMarketContractContextController.tsx

```diff
@@ -0,0 +1,377 @@
+      });
+
+      throw new Error("ERR_PROMPT_MARKET_CONTRACT_INVALID_WALLET_CONNECTION");
+    }
+  };
+
+  const fetchMarketContractValues = async () => {
+    setActions((prev) => ({
+      ...prev,
+      fetchMarketContractValues: {
+        isLoading: true,
+      },
+    }));
+
+    try {
+      // Wait 1 second to allow flags to change
+      setTimeout(async () => {
+        try {
+          if (!signer) throw new Error("not signed in");
+
+          const contract = await connect(DEPLOYED_CONTRACT_ADDRESS, signer);
+
+          const [
+            market,
+            resolution,
+            fees,
+            management,
+            collateralToken,
+            // outcomeIds,
+            // isResolved,
+            // isOpen,
+            // isOver,
+            // isRevealWindowExpired,
+            // isResolutionWindowExpired,
+            // isExpiredUnresolved,
+          ] = await Promise.all([
+            contract.get_market_data(),
+            contract.get_resolution_data(),
+            contract.get_fees_data(),
+            contract.get_management_data(),
+            contract.get_collateral_token_data(),
+            // contract.get_outcome_ids(),
+            // contract.is_resolved(),
+            // contract.is_open(),
+            // contract.is_over(),
+            // contract.is_reveal_window_expired(),
+            // contract.is_resolution_window_expired(),
+            // contract.is_expired_unresolved(),
+          ]);
+
+          const values: PromptWarsMarketContractValues = {
+            market,
+            resolution,
+            fees,
+            management,
+            collateralToken,
+            // TODO:
+            outcomeIds: [],
+            isResolved: false,
+            isOpen: false,
+            isOver: false,
+            isRevealWindowExpired: false,
+            isResolutionWindowExpired: false,
+            isExpiredUnresolved: false,
+            status: PromptWarsMarketContractStatus.LOADING,
+          };
+
+          const status = getMarketStatus(values);
+
+          values.status = status;
+
+          setMarketContractValues(values);
+        } catch (error) {
+          console.log(error);
+        }
+      }, 1000);
+    } catch {
+      toast.trigger({
+        variant: "error",
+        withTimeout: true,
+        title: "Failed to fetch market data",
+        children: <Typography.Text>Try refreshing the page, or check your internet connection.</Typography.Text>,
+      });
+    }
+
+    setActions((prev) => ({
+      ...prev,
+      fetchMarketContractValues: {
+        isLoading: false,
+      },
+    }));
+  };
+
+  // const getOutcomeToken = async (args: GetOutcomeTokenArgs) => {
+  //   try {
+  //     if (!signer) throw new Error("not signed in");
+
+  //     if (!marketContract) {
+  //       marketContract = await await connect(DEPLOYED_CONTRACT_ADDRESS, signer);
+  //     }
+
+  //     const outcomeToken = await marketContract.get_outcome_token(args);
```

Comment:
No get_outcome_token or outcome token

Reply:
Yeah, the `OutcomeToken` in Solidity is now a `Player`.